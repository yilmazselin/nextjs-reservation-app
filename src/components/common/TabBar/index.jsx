import styled from "styled-components";
import { useTranslation } from "next-i18next";
import React, { memo } from "react";
import Link from "next/link";
import { colors, fontSize, fontWeight, padding } from "../../../utils/styles";
import Icon from "../Icon";
import Text from "../Text";

const Container = styled.div`
  background-color: #fafafa;
  width: 100%;
  height: 80px;
  position: fixed;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
`;

const ButtonWrapper = styled.a`
  display: flex;
  flex-direction: column;
  align-items: center;
  cursor: pointer;
`;

function TabBar() {
  const { t } = useTranslation("common");

  return (
    <Container>
      <Link href="/list">
        <ButtonWrapper data-testid="tab-list">
          <Icon color="#4D4D4D" icon="list" size="18" />
          <Text
            color={colors.dark}
            fontSize={fontSize.sm}
            fontWeight={fontWeight.semiBold}
            params={{
              marginTop: padding.xs,
            }}
          >
            {t("common:tabbar.list")}
          </Text>
        </ButtonWrapper>
      </Link>

      <Link href="/account">
        <ButtonWrapper data-testid="tab-account">
          <Icon color="#4D4D4D" icon="person" size="18" />
          <Text
            color={colors.dark}
            fontSize={fontSize.sm}
            fontWeight={fontWeight.semiBold}
            params={{
              marginTop: padding.xs,
            }}
          >
            {t("common:tabbar.account")}
          </Text>
        </ButtonWrapper>
      </Link>
    </Container>
  );
}

export default memo(TabBar);
