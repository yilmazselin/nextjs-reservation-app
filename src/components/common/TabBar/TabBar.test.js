import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import { useTranslation } from "react-i18next";
import * as nextRouter from "next/router";
import "jest-styled-components";
import TabBar from ".";

jest.mock("react-i18next", () => ({
  useTranslation: jest.fn(),
}));
const tSpy = jest.fn((str) => str);

useTranslation.mockReturnValue({
  t: tSpy,
  i18n: {
    changeLanguage: () => new Promise(() => {}),
  },
});

jest.mock(
  "next/link",
  () =>
    ({ children }) =>
      children
);
nextRouter.useRouter = jest.fn();
nextRouter.useRouter.mockImplementation(() => ({ route: "/account" }));

describe("<TabBar />", () => {
  it("should render correctly", () => {
    const { container } = render(<TabBar />);
    expect(container).toMatchSnapshot();
  });
  it("should render csdsdsorrectly", () => {
    const push = jest.fn();

    render(<TabBar />);
    const mytest = screen.getByTestId("tab-account");
    fireEvent.click(mytest);
  });
});
