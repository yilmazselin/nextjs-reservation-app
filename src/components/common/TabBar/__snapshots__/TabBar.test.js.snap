// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabBar /> should render correctly 1`] = `
.c2 {
  color: #121212;
  font-size: 12px;
  font-family: "NunitoSans-600","sans-serif";
  margin-top: 8px;
}

.c0 {
  background-color: #fafafa;
  width: 100%;
  height: 80px;
  position: fixed;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
      data-testid="tab-list"
    >
      <svg
        color="#4D4D4D"
        data-testid="svg-icon"
        style="display: inline-block; stroke: currentColor; fill: currentColor;"
        viewBox="0 0 1024 1024"
      >
        <path
          d="M568.889 0v341.333h455.111v-341.333h-455.111zM568.889 1024h455.111v-568.889h-455.111v568.889zM0 1024h455.111v-341.333h-455.111v341.333zM0 568.889h455.111v-568.889h-455.111v568.889z"
        />
      </svg>
      <p
        class="c2"
        color="#121212"
        font-size="12"
        font-weight="600"
      >
        common:tabbar.list
      </p>
    </a>
    <a
      class="c1"
      data-testid="tab-account"
    >
      <svg
        color="#4D4D4D"
        data-testid="svg-icon"
        style="display: inline-block; stroke: currentColor; fill: currentColor;"
        viewBox="0 0 1024 1024"
      >
        <path
          d="M512 0c67.895 0 133.009 26.971 181.018 74.981 48.013 48.009 74.982 113.124 74.982 181.019s-26.97 133.010-74.982 181.018c-48.009 48.013-113.122 74.982-181.018 74.982s-133.010-26.97-181.019-74.982c-48.009-48.008-74.981-113.122-74.981-181.018s26.971-133.010 74.981-181.019c48.009-48.009 113.124-74.981 181.019-74.981zM512 640c282.88 0 512 114.56 512 256v128h-1024v-128c0-141.44 229.12-256 512-256z"
        />
      </svg>
      <p
        class="c2"
        color="#121212"
        font-size="12"
        font-weight="600"
      >
        common:tabbar.account
      </p>
    </a>
  </div>
</div>
`;
